src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -405,6 +406,46 @@ static void M()
             Assert.Equal(attrType, info.Symbol);
         }
 
+        [Theory]
+        [InlineData("[A]")]
+        [InlineData("[return: A]")]
+        public void LocalFunctionAttribute_SpeculativeSemanticModel(string attributes)
+        {
+            var text = $@"
+using System;
+
+class A : Attribute {{ }}
+
+class C
+{{
+    void M()
+    {{
+        {attributes} void local1() {{ }}
+    }}
+}}
+";
+            var tree = SyntaxFactory.ParseSyntaxTree(text, TestOptions.RegularPreview);
+            var model = CreateCompilation(tree).GetSemanticModel(tree);
+            var attrClass = model
+                .GetDeclaredSymbol(tree.FindNodeOrTokenByKind(SyntaxKind.ClassDeclaration).AsNode())
+                .GetSymbol<NamedTypeSymbol>();
+            Assert.Equal("A", attrClass.Name);
+
+            var attrSyntax = (AttributeSyntax)tree.FindNodeOrTokenByKind(SyntaxKind.Attribute).AsNode();
+            var attrSymbol = model.GetSymbolInfo(attrSyntax.Name).Symbol.GetSymbol<MethodSymbol>();
+            Assert.Equal(attrClass.Constructors.Single(), attrSymbol);
+
+            // Verify with speculative semantic model
+            var newTree = SyntaxFactory.ParseSyntaxTree(text + " ", TestOptions.RegularPreview);
+            var newMethod = newTree.GetRoot().DescendantNodes().OfType<MethodDeclarationSyntax>().Single();
+            Assert.True(model.TryGetSpeculativeSemanticModelForMethodBody(newMethod.Body.SpanStart, newMethod, out var speculativeModel));
```

Comment:
Please actually insert new code in the attribute and use the speculative model to query info about it.

Reply:
Spoke last Friday about this. We agree that this test is using the speculative model to get a symbol for the attribute, correct?